### DISCUSSION TITLE
How to add google analytics

### DISCUSSION THREAD

**Discussion by mazdak66** [FreshRSS/FreshRSS#7005] [Q&A]
Is there a way to add google analytics or tag manager to self hosted instance?
If yes, can you please let me know which file exactly needs to be modified?

Thanks,

**Comment by Alkarex** [comment]
By design, the system sends CSP HTTP headers to avoid such spyware and other third-party injections.
https://developer.mozilla.org/docs/Web/HTTP/CSP

The easiest is to self-host your stats.
However, you can modify the CSP headers to decrease the security at your Web server level, or proxy level, or by using the `Minz_Extension::amendCsp()` function in  the case of a FreshRSS extension.

To add the third-part JavaScript code, you can manually edit
https://github.com/FreshRSS/FreshRSS/blob/5b28a35003a015e29770094932157f13a3f7f5c0/app/layout/layout.phtml
but this will be overridden at each new version of FreshRSS, so it would better to make a FreshRSS extension on the model of 
https://github.com/FreshRSS/FreshRSS/blob/1c09408c6459eb8d719d94ba593edfa44883cb85/lib/core-extensions/UserJS/extension.php
